src/Workspaces/Core/Portable/NamingStyles/NamingStyleRules.cs

```diff
@@ -46,5 +59,43 @@ private bool IsSymbolNameAnalyzable(ISymbol symbol)
 
             return true;
         }
+
+        private bool DoesSymbolImplementAnotherSymbol(ISymbol symbol)
+        {
+            if (symbol.IsStatic || symbol.DeclaredAccessibility != Accessibility.Public)
+            {
+                return false;
+            }
+
+            return symbol.IsOverride ||
+                symbol.ExplicitInterfaceImplementations().Length > 0 ||
+                IsInterfaceImplementation(symbol);
+        }
+
+        /// <summary>
+        /// This does not handle the case where a method in a base type implicitly implements an
+        /// interface method on behalf of one of its derived types.
+        /// </summary>
+        private bool IsInterfaceImplementation(ISymbol symbol)
+        {
+            var containingType = symbol.ContainingType;
+            var implementedInterfaces = containingType.AllInterfaces;
+
+            for (var i = 0; i < implementedInterfaces.Length; i++)
+            {
+                var implementedInterface = implementedInterfaces[i];
+                var implementedInterfaceMembersWithSameName = implementedInterface.GetMembers(symbol.Name);
+                for (var j = 0; j < implementedInterfaceMembersWithSameName.Length; j++)
+                {
+                    var implementedInterfaceMember = implementedInterfaceMembersWithSameName[j];
+                    if (containingType.FindImplementationForInterfaceMember(implementedInterfaceMember).Equals(symbol))
```

Comment:
FindImplementationForInterfaceMember returns null when it fails, no?  So this should be symbol.Equals(...) instead.